src/Console/ConsoleInitializer.php

```diff
@@ -6,18 +6,16 @@
 
 use Tempest\Container\Container;
 use Tempest\Container\Initializer;
+use Tempest\Container\Singleton;
 
+#[Singleton]
 class ConsoleInitializer implements Initializer
 {
-    public function initialize(Container $container): object
+    public function initialize(Container $container): Console
     {
-        $console = new GenericConsole(
+        return new GenericConsole(
             $container->get(ConsoleInput::class),
             $container->get(ConsoleOutput::class),
         );
```

Comment:
To me, this is a good example of where having initializer methods make sense. For example:

```php
class ConsoleInitializer
{
    #[Singleton]
    #[Initializes(ConsoleInput::class)]
    #[Initializes(GenericConsoleInput::class)]
    public function initializeConsoleInput(): ConsoleInput
    {
        return new GenericConsoleInput(
            output: $this->container->get(ConsoleOutput::class)
        );
    }

    #[Singleton]
    #[Initializes(
        ConsoleOutput::class,
        GenericConsoleOutput::class
    )] // Alternative syntax.
    public function initializeConsoleOutput(): ConsoleOutput
    {
        return new GenericConsoleOutput();
    }
}
```

Reply:
Right now I prefer the other approach better, but maybe we should reconsider it closer to 1.0